lib/phoenix_inline_svg/helpers.ex

```diff
@@ -264,17 +219,34 @@ defmodule PhoenixInlineSvg.Helpers do
     |> Enum.into([])
   end
 
-  defp create_cached_svg_image({collection, name}, svgs_path) do
-    filename = name |> String.split(".") |> List.first
+  defp create_cached_svg_image({collection, name}) do
+    filename =
+      name
+      |> String.split("/")
+      |> List.last
+      |> String.split(".") 
+      |> List.first
+    svg = read_svg_from_path(name)
 
     quote do
+      def svg_image(unquote(filename)) do
+        unquote(PhoenixInlineSvg.Utils.safety_string(svg))
+      end
+
       def svg_image(unquote(filename), unquote(collection)) do
-        unquote(
-          [svgs_path, collection, name]
-          |> Path.join
-          |> read_svg_from_path
-          |> safety_string
-        )
+        unquote(PhoenixInlineSvg.Utils.safety_string(svg))
+      end
+
+      def svg_image(unquote(filename), opts) when is_list(opts) do
```

Comment:
The next def will call the same functions the same way regardless of if opts is a list. This def seems to be redundant because of that (or the other one shouldn't be there).

Reply:
The functions themselves are both necessary. But I am able to have the lower 1 and 2 arity functions delegate to `svg_image/3`